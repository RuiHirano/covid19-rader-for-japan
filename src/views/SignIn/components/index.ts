export { default as BackButton } from './BackButton/BackButton';
export { default as ImageField } from './ImageField/ImageField';
export { default as SignInForm } from './SignInForm';