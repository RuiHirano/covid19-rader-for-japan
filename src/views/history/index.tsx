import React, { useState, useEffect } from "react";
import { makeStyles, Theme } from "@material-ui/core/styles";

import { HistoryToolbar, HistoryTable } from "./components";
import { useSelector } from "react-redux";
import { LoadingState, Item } from "../../types";
import { ReduxState } from "../../redux/module";
import { Main as MainLayout } from "../../layouts";

// Container
interface ContainerProps { }
const HistoryContainer: React.FC<ContainerProps> = props => {
    /*const dispatch = useDispatch();
    const handleUpdateHisyory = (values: FormikValues) => {
        dispatch(
            userActions.updateHisyoryAction({ password: values.password })
        );
	};*/
    const items_ = useSelector((state: ReduxState) => state.Items);
    const [items, setItems] = useState(items_);

    return <HistoryView items={items} />;
};

export default HistoryContainer;

interface Props {
    items: Item[];
}

const HistoryView: React.FC<Props> = props => {
    const { items } = props;
    const classes = useStyles();

    //const items = useSelector((state: ReduxState) => state.Items);

    return (

        <MainLayout title="History">
            <HistoryToolbar />
            <div className={classes.content}>
                <HistoryTable items={items} />
            </div>

        </MainLayout>
    );
};

const useStyles = makeStyles((theme: Theme) => ({
    root: {
        padding: theme.spacing(3)
    },
    content: {
        marginTop: theme.spacing(2)
    }
}));
