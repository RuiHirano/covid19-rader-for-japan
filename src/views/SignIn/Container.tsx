import React, { useState, useEffect, useRef } from "react";
import { connect } from "react-redux";
import { Dispatch, Action } from "redux";
import SignIn from "./SignIn";
import { AppState, ActionTypes } from "./../../redux/configureStore";
import { SignInState, signActions } from "../../redux/saga/Sign/signSaga";
import { withRouter, RouteComponentProps, match } from "react-router";
import { Loading, LoadingState } from "../../types";
import * as H from "history";

interface Props {
    handleSignIn: (signInState: SignInState) => void;
    isLoading: Loading["IsLoading"];
    loadingState: Loading["LoadingState"];
    history: H.History;
    location: H.Location;
    match: match;
}

function usePrevious(value: any) {
    const ref = useRef();
    useEffect(() => {
        ref.current = value;
    });
    return ref.current;
}

const SignInContainer: React.FC<Props> = props => {
    const { history, handleSignIn, isLoading, loadingState } = props;
    const prevIsLoading = usePrevious(isLoading);

    const handleBack = () => {
        history.goBack();
    };

    useEffect(() => {
        console.log("useEffect", isLoading, loadingState);
        if (
            prevIsLoading &&
            !isLoading &&
            loadingState == LoadingState.SIGN_IN
        ) {
            history.push("/dashboard");
        }
    }, [isLoading]);

    return <SignIn handleBack={handleBack} handleSignIn={handleSignIn} />;
};

function mapStateToProps(state: AppState) {
    const loading = state.App.Loading;
    return {
        isLoading: loading.IsLoading,
        loadingState: loading.LoadingState
    };
}

function mapDispatchToProps(dispatch: Dispatch<Action>) {
    return {
        handleSignIn: (signInState: SignInState) =>
            dispatch(signActions.signInAction(signInState))
    };
}

export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps
    )(SignInContainer)
);
