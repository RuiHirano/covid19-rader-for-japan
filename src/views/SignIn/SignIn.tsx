import React, { useEffect, useRef } from "react";
import { withRouter, match } from "react-router-dom";
import { makeStyles, Theme } from "@material-ui/core/styles";
import { SignInState, signActions } from "../../redux/saga/Sign/signSaga";
import { Grid } from "@material-ui/core";
import ImageField from "./components/ImageField/ImageField";
import BackButton from "./components/BackButton/BackButton";
import SignInForm from "./components/SignInForm/SignInForm";

import { useDispatch, useSelector } from "react-redux";
import { AppState } from "../../redux/configureStore";
import { Loading, LoadingState } from "../../types";
import * as H from "history";
import { useLoading } from "../../common/hooks/useLoading";

// Container
interface ContainerProps {
    history: H.History;
    location: H.Location;
    match: match;
}

const SignInContainer: React.FC<ContainerProps> = props => {
    const { history } = props;

    const { isLoading, isFinishLoading } = useLoading(LoadingState.SIGN_IN);

    useEffect(() => {
        if (isFinishLoading) {
            history.push("/dashboard");
        }
    }, [isLoading]);

    return <SignIn />;
};

export default withRouter(SignInContainer);

// Presentational
interface Props {}

const SignIn: React.FC<Props> = props => {
    const classes = useStyles();

    return (
        <div className={classes.root}>
            <Grid className={classes.grid} container>
                <Grid className={classes.quoteContainer} item lg={5}>
                    <ImageField />
                </Grid>
                <Grid className={classes.content} item lg={7} xs={12}>
                    <div className={classes.contentHeader}>
                        <BackButton />
                    </div>
                    <div className={classes.contentBody}>
                        <SignInForm />
                    </div>
                </Grid>
            </Grid>
        </div>
    );
};

const useStyles = makeStyles((theme: Theme) => ({
    root: {
        backgroundColor: theme.palette.background.default,
        height: "100%"
    },
    grid: {
        height: "100%"
    },
    quoteContainer: {
        [theme.breakpoints.down("md")]: {
            display: "none"
        }
    },
    content: {
        height: "100%",
        display: "flex",
        flexDirection: "column"
    },
    contentHeader: {
        display: "flex",
        alignItems: "center",
        paddingTop: theme.spacing(5),
        paddingBototm: theme.spacing(2),
        paddingLeft: theme.spacing(2),
        paddingRight: theme.spacing(2)
    },
    contentBody: {
        flexGrow: 1,
        display: "flex",
        alignItems: "center",
        [theme.breakpoints.down("md")]: {
            justifyContent: "center"
        }
    }
}));

//export default SignIn;
