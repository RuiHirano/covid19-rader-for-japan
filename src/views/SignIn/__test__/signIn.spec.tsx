import * as React from "react";
import renderer from "react-test-renderer";
import { configure, shallow, mount } from "enzyme";
import SignIn from "../SignIn";
import SignInContainer from "../SignInContainer";
import {
    cleanup,
    fireEvent,
    getByLabelText,
    getByTestId,
    act,
    render
} from "@testing-library/react";
import {
    BackButton,
    EmailField,
    PasswordField,
    ImageField,
    SignInForm,
    SocialSignIn,
    SubmitButton,
    Title,
    ToSignUp
} from "../components";
import {
    FormikErrors,
    FormikValues,
    FormikTouched,
    FormikHandlers,
    Formik
} from "formik";
import { Route, Router } from "react-router";
import { createMemoryHistory } from "history";

interface Props {
    errors: FormikErrors<FormikValues>;
    touched: FormikTouched<FormikValues>;
    values: {
        email: string;
        password: string;
    };
    handleChange: FormikHandlers["handleChange"];
    handleBlur: FormikHandlers["handleBlur"];
}

describe("スクリーンショットテスト", () => {
    const handleSignIn = jest.fn();
    const handleBack = jest.fn();
    const handleSubmit = jest.fn();

    const props: Props = {
        errors: {},
        touched: {},
        values: {
            email: "",
            password: ""
        },
        handleChange: jest.fn(),
        handleBlur: jest.fn()
    };

    it("Snapshot of <SignIn />", () => {
        const signIn = shallow(
            <SignIn handleSignIn={handleSignIn} handleBack={handleBack} />
        );
        expect(signIn).toMatchSnapshot();
    });

    it("Snapshot of <SignInContainer />", () => {
        const signInContainer = shallow(<SignInContainer />);
        expect(signInContainer).toMatchSnapshot();
    });

    it("Snapshot of <BackButton />", () => {
        const backButton = shallow(<BackButton handleBack={handleBack} />);
        expect(backButton).toMatchSnapshot();
    });

    it("Snapshot of <EmailField />", () => {
        const emailField = shallow(<EmailField {...props} />);
        expect(emailField).toMatchSnapshot();
    });

    it("Snapshot of <PasswordField />", () => {
        const passwordField = shallow(<PasswordField {...props} />);
        expect(passwordField).toMatchSnapshot();
    });

    it("Snapshot of <ImageField />", () => {
        const imageField = shallow(<ImageField {...props} />);
        expect(imageField).toMatchSnapshot();
    });

    it("Snapshot of <SignInForm />", () => {
        const signInForm = shallow(<SignInForm handleSignIn={handleSignIn} />);
        expect(signInForm).toMatchSnapshot();
    });

    it("Snapshot of <SocialSignIn />", () => {
        const socialSignIn = shallow(<SocialSignIn />);
        expect(socialSignIn).toMatchSnapshot();
    });

    it("Snapshot of <SubmitButton />", () => {
        const submitButton = shallow(
            <SubmitButton handleSubmit={handleSubmit} />
        );
        expect(submitButton).toMatchSnapshot();
    });

    it("Snapshot of <Title />", () => {
        const title = shallow(<Title />);
        expect(title).toMatchSnapshot();
    });

    it("Snapshot of <ToSignUp />", () => {
        const toSignUp = shallow(<ToSignUp />);
        expect(toSignUp).toMatchSnapshot();
    });
});

describe("機能テスト", () => {
    const props: Props = {
        errors: {},
        touched: {},
        values: {
            email: "",
            password: ""
        },
        handleChange: jest.fn(),
        handleBlur: jest.fn()
    };

    it.todo("SignUpを押すとSignUp画面へ遷移");
    it.todo("Emailが正しい表記でない場合、エラーメッセージが表示される");
    it.todo("Emailが空白の場合、空白エラーメッセージが表示される");
    it.todo(
        "Pasdswordが正しい表記(a-z0-9)でない場合、エラーメッセージが表示される"
    );
    it.todo("Pasdswordが8文字以下場合、エラーメッセージが表示される");

    it("Passwordが空白の場合、空白エラーメッセージが表示される", () => {
        const handleSignIn = jest.fn();
        const wrapper = shallow(<SignInForm handleSignIn={handleSignIn} />);
        //const wrapper = shallow(<PasswordField {...props} />);
        const passwordField = wrapper
            .find(Formik)
            .dive()
            .find(PasswordField);
        console.log("test", passwordField.props());
    });

    it("正しい入力でSignUpNowボタンを押すとhandleSignUpに正しい引数が渡される", () => {
        // FIX:
        const values: Props["values"] = {
            email: "test@test.com",
            password: "testtesttest"
        };

        const expectValues: Props["values"] = {
            email: "test@test.com",
            password: "testtesttest"
        };

        const handleSignIn = jest.fn();
        const history = createMemoryHistory();
        const { getByText } = render(
            <Router history={history}>
                <SignInForm handleSignIn={handleSignIn} />
            </Router>
        );

        const a = getByText("adsf");
        //		wrapper.find(Formik).simulate("change", {
        //            target: { initialValues: { ...values, email: "testtest" } }
        //        });
        //        wrapper.find(Formik).simulate("submit", values);

        expect(handleSignIn.mock.calls).toEqual([[expectValues]]);
    });
    it.todo(
        "正しくない入力でSignUpNowボタンを押すとhandleSignUpに正しい引数が渡されない"
    );
    it.todo("BackButtonを押すとhandleBackButtonが呼ばれる");
    it.todo("LoginFacebookボタンを押すとfacebook関数が呼ばれる");
    it.todo("LoginGoogleボタンを押すとgoogle関数が呼ばれる");
    // state変化
    it.todo("SignInNowを押すとLoading=Trueになりボタンが押せなくなる");
    it.todo("LoadingがTrue->False, StatusがSIGN_INの場合画面遷移する");
});
