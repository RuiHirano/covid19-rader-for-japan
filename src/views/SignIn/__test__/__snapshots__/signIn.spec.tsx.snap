// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スクリーンショットテスト Snapshot of <BackButton /> 1`] = `
<WithStyles(ForwardRef(IconButton))
  onClick={[MockFunction]}
>
  <ArrowBackIcon />
</WithStyles(ForwardRef(IconButton))>
`;

exports[`スクリーンショットテスト Snapshot of <EmailField /> 1`] = `
<WithStyles(ForwardRef(TextField))
  className="makeStyles-textField-21"
  error={false}
  fullWidth={true}
  helperText={null}
  label="Email"
  name="Email"
  type="text"
  value=""
  variant="outlined"
/>
`;

exports[`スクリーンショットテスト Snapshot of <ImageField /> 1`] = `
<div
  className="makeStyles-quote-24"
>
  <div
    className="makeStyles-quoteInner-25"
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-quoteText-26"
      variant="h1"
    >
      Hella narwhal Cosby sweater McSweeney's, salvia kitsch before they sold out High Life.
    </WithStyles(ForwardRef(Typography))>
    <div>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-name-27"
        variant="body1"
      >
        Takamaru Ayako
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-bio-28"
        variant="body2"
      >
        Manager at inVision
      </WithStyles(ForwardRef(Typography))>
    </div>
  </div>
</div>
`;

exports[`スクリーンショットテスト Snapshot of <PasswordField /> 1`] = `
<WithStyles(ForwardRef(TextField))
  className="makeStyles-textField-22"
  error={false}
  fullWidth={true}
  helperText={null}
  label="Password"
  name="Password"
  type="password"
  value=""
  variant="outlined"
/>
`;

exports[`スクリーンショットテスト Snapshot of <SignIn /> 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-grid-2"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-quoteContainer-3"
      item={true}
      lg={5}
    >
      <ImageField />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-content-10"
      item={true}
      lg={7}
      xs={12}
    >
      <div
        className="makeStyles-contentHeader-11"
      >
        <BackButton
          handleBack={[MockFunction]}
        />
      </div>
      <div
        className="makeStyles-contentBody-13"
      >
        <SignInForm
          handleSignIn={[MockFunction]}
        />
      </div>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`スクリーンショットテスト Snapshot of <SignInContainer /> 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`スクリーンショットテスト Snapshot of <SignInForm /> 1`] = `
<Formik
  initialValues={
    Object {
      "email": "",
      "password": "",
    }
  }
  onSubmit={[Function]}
  validationSchema={
    ObjectSchema {
      "_blacklist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "_conditions": Array [],
      "_defaultDefault": [Function],
      "_deps": Array [],
      "_excludedEdges": Array [],
      "_exclusive": Object {},
      "_mutate": undefined,
      "_nodes": Array [
        "password",
        "email",
      ],
      "_options": Object {
        "abortEarly": true,
        "recursive": true,
      },
      "_type": "object",
      "_typeError": [Function],
      "_whitelist": RefSet {
        "list": Set {},
        "refs": Map {},
      },
      "fields": Object {
        "email": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "required": true,
            "undefined": false,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
        "password": StringSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_deps": Array [],
          "_exclusive": Object {
            "min": true,
            "required": true,
            "undefined": false,
          },
          "_mutate": undefined,
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "string",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "tests": Array [
            [Function],
            [Function],
            [Function],
          ],
          "transforms": Array [
            [Function],
          ],
        },
      },
      "tests": Array [],
      "transforms": Array [
        [Function],
      ],
    }
  }
>
  <Component />
</Formik>
`;

exports[`スクリーンショットテスト Snapshot of <SocialSignIn /> 1`] = `
<div>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-socialButtons-30"
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
        size="large"
        variant="contained"
      >
        <Facebook
          className="makeStyles-socialIcon-31"
        />
        Login with Facebook
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(Button))
        onClick={[Function]}
        size="large"
        variant="contained"
      >
        <Google
          className="makeStyles-socialIcon-31"
        />
        Login with Google
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Typography))
    align="center"
    className="makeStyles-sugestion-32"
    color="textSecondary"
    variant="body1"
  >
    or login with email address
  </WithStyles(ForwardRef(Typography))>
</div>
`;

exports[`スクリーンショットテスト Snapshot of <SubmitButton /> 1`] = `
<WithStyles(ForwardRef(Button))
  className="makeStyles-signInButton-33"
  color="primary"
  disabled={false}
  fullWidth={true}
  onClick={[Function]}
  size="large"
  type="submit"
  variant="contained"
>
  Sign up now
</WithStyles(ForwardRef(Button))>
`;

exports[`スクリーンショットテスト Snapshot of <Title /> 1`] = `
<div>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-title-34"
    variant="h2"
  >
    Sign in
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    color="textSecondary"
    gutterBottom={true}
  >
    Sign in with social media
  </WithStyles(ForwardRef(Typography))>
</div>
`;

exports[`スクリーンショットテスト Snapshot of <ToSignUp /> 1`] = `
<WithStyles(ForwardRef(Typography))
  color="textSecondary"
  variant="body1"
>
  Don't have an account?
   
  <WithStyles(ForwardRef(Link))
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Link",
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    to="/sign-up"
    variant="h6"
  >
    Sign up
  </WithStyles(ForwardRef(Link))>
</WithStyles(ForwardRef(Typography))>
`;
